pymatviz/data/regression.py

```diff
@@ -0,0 +1,11 @@
+"""Dummy regression data for testing and prototyping."""
+
+import numpy as np
+
+
+n_samples = 500
+np_rng = np.random.default_rng(seed=0)
+
+y_true = np_rng.normal(5, 4, n_samples)
+y_pred = 1.2 * y_true - 2 * np_rng.normal(0, 1, n_samples)
+y_std = (y_true - y_pred) * 10 * np_rng.normal(0, 0.1, n_samples)
```

Comment:
I'm feeling this might be too "magic" to be made public (everything about it is hard coded), either we put it under `util` for tests, or perhaps make it more complete like `generate_random_regression_data`?

Reply:
excellent point! i was also thinking about breaking up `utils`. not sure about `utils.testing` but i'm all for having a function `pymatviz.data.random_regression()` similar to `plotly.express.data.tips()`

Reply:
Great I guess I would prefer launch a separate PR and finish that before this one, currently this PR is already overly large.